### ISSUE TITLE
dap-repl buffer does not automatically close

### ISSUE THREAD

**Issue by dohmjan** [rcarriga/cmp-dap#7]
Hello!
I am using `nvim-dap` with your `nvim-dap-ui` and lately also tried `cmp-dap`. I really like it, thank you!

Without `cmp-dap`, the repl buffer automatically closes when toggling/closing the dap-ui. The corresponding issue was raised [here](https://github.com/rcarriga/nvim-dap-ui/issues/165). Your workaround [in this comment](https://github.com/rcarriga/nvim-dap-ui/issues/165#issuecomment-1377712562) and your [PR in nvim-dap](https://github.com/mfussenegger/nvim-dap/pull/839) both worked.
Now, after I included `cmp-dap`, the issue reappears. Could you imagine whether this is due to `cmp-dap`?

**Comment by rcarriga** [comment]
By not closed do you mean it is still listed (i.e. appears in `:ls`)?

There's nothing in cmp-dap that would be do that, but perhaps nvim-cmp is doing something? I do experience any issue with it however

**Comment by jaghaimo** [comment]
Yes. It doesn't exist until I select something from autocompletion in repl, then:
![image](https://user-images.githubusercontent.com/1764586/230896355-e4d13249-c9bd-45d0-97e4-b395d10ffaca.png)

Finished the session and it's still there.

**Comment by oshevtsov** [comment]
I encountered the same behavior and could not find a quick fix. The temporary workaround I am using at the moment is to disable `cmp-dap` for `[dap-repl]` buffers (comment out `"dap-repl"` when setting up `require("cmp").setup.filetype(...)`) and rely on `nvim-dap`'s completion suggestion via `<C-x><C-o>` (built-in omnifunc implementation).

**Comment by EasonMo** [comment]
@dohmjan @oshevtsov @jaghaimo Hi, I have got a solution to hide dap-repl buffer, **not close**. Because this buffer shoud be reuse by the next dap session. It can be checked at my config files [MozyVim](https://github.com/EasonMo/MozyVim).
```lua
local dap = require("dap")
local function clear_dap_ui()
  local bufs = utils.find_buffers_by_filetype("dap-repl")
  for _, buf in ipairs(bufs) do
    vim.bo[buf].buflisted = false
  end
end
dap.listeners.before.event_terminated["dapui_config"] = function()
  clear_dap_ui()
end
dap.listeners.after.disconnect["dapui_config"] = function()
  clear_dap_ui()
end
```
